spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/AbstractRabbitAnnotationDrivenTests.java

```diff
@@ -260,6 +266,50 @@ public void defaultHandle(@Validated String msg) {
 		}
 	}
 
+	/**
+	 * Test for {@link RabbitListenerRepeatableBean} and {@link RabbitListenersBean} that validates that the
+	 * {@code @RabbitListener} annotation is repeatable and generate one specific container per annotation.
+	 */
+	public void testRabbitListenerRepeatable(ApplicationContext context) {
+		RabbitListenerContainerTestFactory simpleFactory =
+				context.getBean("rabbitListenerContainerFactory", RabbitListenerContainerTestFactory.class);
+		assertEquals(2, simpleFactory.getListenerContainers().size());
+
+		MethodRabbitListenerEndpoint first = (MethodRabbitListenerEndpoint)
+				simpleFactory.getListenerContainer("first").getEndpoint();
+		assertEquals("first", first.getId());
+		assertEquals("myQueue", first.getQueueNames().iterator().next());
+
+		MethodRabbitListenerEndpoint second = (MethodRabbitListenerEndpoint)
+				simpleFactory.getListenerContainer("second").getEndpoint();
+		assertEquals("second", second.getId());
+		assertEquals("anotherQueue", second.getQueueNames().iterator().next());
+	}
+
+	@Component
+	static class RabbitListenerRepeatableBean {
+
+		@RabbitListener(id = "first", queues = "myQueue")
+		@RabbitListener(id = "second", queues = "anotherQueue")
+		public void repeatableHandle(String msg) {
```

Comment:
@artembilan @wilkinsona This will break the Spring IO JDK 7 compatibility build.

I guess we have two options; disable the JDK7 build or comment out this test. I think I'm leaning towards the latter since the class below is functionally equivalent.

Thoughts?